src/main/java/uk/gov/hmcts/reform/unspec/service/tasks/handler/PollingEventEmitterHandler.java

```diff
@@ -18,22 +16,23 @@
 @RequiredArgsConstructor
 @Component
 @ConditionalOnExpression("${polling.event.emitter.enabled:true}")
-public class PollingEventEmitterHandler implements ExternalTaskHandler {
+public class PollingEventEmitterHandler implements BaseExternalTaskHandler {
 
     private final CaseReadyBusinessProcessSearchService caseSearchService;
     private final CaseDetailsConverter caseDetailsConverter;
     private final EventEmitterService eventEmitterService;
 
     @Override
-    public void execute(ExternalTask externalTask, ExternalTaskService externalTaskService) {
-        final String taskName = externalTask.getTopicName();
-
+    public void handleTask(ExternalTask externalTask) {
         List<CaseDetails> cases = caseSearchService.getCases();
-        log.info("Job '{}' found {} case(s)", taskName, cases.size());
+        log.info("Job '{}' found {} case(s)", externalTask.getTopicName(), cases.size());
         cases.stream()
             .map(caseDetailsConverter::toCaseData)
             .forEach(eventEmitterService::emitBusinessProcessCamundaEvent);
+    }
 
-        externalTaskService.complete(externalTask);
+    @Override
+    public int getMaxRetries() {
+        return 1;
```

Comment:
PollingEventEmitter should have no retries as you pointed out. I assume it's indicating number of attempts then, so maybe we should call it `getMaxAttempts`?

Reply:
yes, retries gets decereased by 1 in the handleFailure code. attempts is a better name for this, thanks 👍